src/txn/txn_ckpt.c

```diff
@@ -1863,12 +1864,13 @@ __wt_checkpoint_close(WT_SESSION_IMPL *session, bool final)
 
 /*
  * __checkpoint_timing_stress --
- *     Optionally add a 10 second delay to a checkpoint to simulate a long running checkpoint for
- *     debug purposes. The reason for this option is finding operations that can block while waiting
- *     for a checkpoint to complete.
+ *     Optionally add a delay to a checkpoint to simulate a long running checkpoint for debug
+ *     purposes. The reason for this option is finding operations that can block while waiting for a
+ *     checkpoint to complete.
  */
 static void
-__checkpoint_timing_stress(WT_SESSION_IMPL *session, bool history_store_stress)
+__checkpoint_timing_stress(
+  WT_SESSION_IMPL *session, uint64_t flag, uint64_t seconds, uint64_t micro_seconds)
```

Comment:
This should take a `struct timespec` parameter, rather than two different timers. Similar to `__time_check_monotonic`. I know that means the callers probably need a local variable, but I think it's worth it.

I *think* `__wt_sleep` doesn't do that because it's used external to the library (or it was), and managing what a timespec is externally is a bit of a pain.

Reply:
Thanks, changes made.